include/nbl/video/surface/ISurface.h

```diff
@@ -72,6 +72,108 @@ class NBL_API ISurface : public core::IReferenceCounted
             EST_FLAG_BITS_MAX_ENUM = 0x7FFFFFFF
         };
 
+        #include "nbl/builtin/glsl/utils/swapchain_transform_e.h"
+        static_assert(int(EST_IDENTITY_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY);
+        static_assert(int(EST_ROTATE_90_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90);
+        static_assert(int(EST_ROTATE_180_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180);
+        static_assert(int(EST_ROTATE_270_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270);
+        static_assert(int(EST_HORIZONTAL_MIRROR_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR);
+        static_assert(int(EST_HORIZONTAL_MIRROR_ROTATE_90_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90);
+        static_assert(int(EST_HORIZONTAL_MIRROR_ROTATE_180_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180);
+        static_assert(int(EST_HORIZONTAL_MIRROR_ROTATE_270_BIT) == NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270);
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180
+        #undef NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270
+
+        // A matrix that can be pre-multiplied to the projection matrix in order to apply the
+        // surface transform.
+        static inline core::matrix4SIMD surfaceTransformForward(const E_SURFACE_TRANSFORM_FLAGS transform)
```

Comment:
Name it: `getSurfaceTransformationMatrix` 

PS. I normally don't give a lot of shit about names, especially the ones used in extensions and everything but since this PR is considered one of the major api changes, we need to do most things right from the beginning.

remember to update README with new names and comments